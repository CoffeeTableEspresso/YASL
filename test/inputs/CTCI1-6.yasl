##a2b1c4a3\nabcde\n

fn compress(s) {
    new_s := ''
    i := 0
    while i < len s {
        count := 0
        j := i
        while i < len s && s[i] == s[j] {
            count += 1
            i += 1
        }
        new_s ~= s[j] ~ count
    }
    if len s <= len new_s {
        return s
    }
    return new_s
}


echo compress('aabccccaaa')
echo compress('abcde')

