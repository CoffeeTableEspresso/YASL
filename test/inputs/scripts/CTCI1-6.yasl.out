a2b1c4a3
abcde
